### PR TITLE
refactor(utilities): refactoring Utilities constants

### PR DESCRIPTION
**Pull request by ssi02014**
## What is the motivation for this pull request?

Hello 👋, @remarkablemark 
I worked on refactoring by isolating code that could be managed by constants inside the `utilities.js`.
Please review this pull request to see if it's appropriate. 🙏

## What is the current behavior?
There is no change in behavior.

## What is the new behavior?
There is no change in behavior.

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [x] Documentation
- [x] Types

### CODE REVIEW

File: lib/utilities.js [hunk 1]
```diff
@@ -13,14 +13,12 @@ function invertObject(obj, override) {
...
-  var key;
-  var value;
   var isOverridePresent = typeof override === 'function';
   var overrides = {};
   var result = {};
 
-  for (key in obj) {
-    value = obj[key];
+  for (var key in obj) {
+    var value = obj[key];
```
Comment (ssi02014) [inline lib/utilities.js:21]: I think it would be clearer to declare it in the for statement block. 🙏

File: lib/utilities.js [hunk 2]
```diff
@@ -47,31 +45,34 @@ function invertObject(obj, override) {
...
-    default:
-      return true;
+  // These are reserved SVG and MathML elements.
+  // We don't mind this whitelist too much because we expect it to never grow.
+  // The alternative is to track the namespace in a few places which is convoluted.
+  // https://w3c.github.io/webcomponents/spec/custom/#custom-elements-core-concepts
+  if (RESERVED_SVG_MATHML_ELEMENTS.has(tagName)) {
+    return false;
   }
+  return true;
```
Comment (ssi02014) [inline lib/utilities.js:71]: I worked similarly to the existing "elementsWithNoTextChildren" approach.  🙏

File: lib/utilities.js [hunk 3]
```diff
@@ -47,31 +45,34 @@ function invertObject(obj, override) {
...
-    case 'missing-glyph':
-      return false;
-    default:
-      return true;
+  // These are reserved SVG and MathML elements.
+  // We don't mind this whitelist too much because we expect it to never grow.
+  // The alternative is to track the namespace in a few places which is convoluted.
+  // https://w3c.github.io/webcomponents/spec/custom/#custom-elements-core-concepts
+  if (RESERVED_SVG_MATHML_ELEMENTS.has(tagName)) {
+    return false;
```
Comment (ssi02014) [inline lib/utilities.js:69]: I worked similarly to the existing "elementsWithNoTextChildren" approach. 🙏

File: lib/utilities.js (hunk 1 above)
Comment (remarkablemark) [reply lib/utilities.js:21]: There is a premature optimization when the variable is initialized once, but it is a minor so I am fine with your change

**Comment by remarkablemark** [comment]
@ssi02014 I noticed you added a fix 9ec5ba1 so I wanted to confirm if that's true since it will trigger a patch version release for this package

**Comment by ssi02014** [comment]
@remarkablemark 
This is a minor fix that incorrectly replaced the existing "elementsWithNoTextChildren" with "ELEMENTS_WITH_TEXT_CHILDREN" and changed it back to "ELEMENTS_WITH_NO_TEXT_CHILDREN". 🙇‍♂️

**Review by remarkablemark — APPROVED** [review]
Thanks @ssi02014! LGTM

**Comment by remarkablemark** [comment]
Release:

- https://github.com/remarkablemark/html-react-parser/pull/1004#issuecomment-1672052169

**Pull request merged**